File: CPP/Elektronik.cpp

#include <iostream>
#include <string>
using namespace std;

class Elektronik{
private:
    int id;
    string nama;
    string merek;
    double harga;

public:
    //konstruktor 
    Elektronik() : id(0), nama(""), merek(""), harga(0.0) {}

    //konstruktor dengan parameter
    Elektronik(int i, string n, string m, double h)
        : id(i), nama(n), merek(m), harga(h) {}

    //getter
    int getId()const{ 
        return id; 
    }
    string getNama() const{ 
        return nama; 
    }
    string getMerek() const{ 
        return merek; 
    }
    double getHarga() const{ 
        return harga; 
    }

    //setter
    void setId(int i){ 
        id = i; 
    }
    void setNama(string n){
        nama = n; 
    }
    void setMerek(string m){ 
        merek = m; 
    }
    void setHarga(double h){ 
        harga = h; 
    }

    //method tampil
    void tampil() const {
        cout << "ID: " << id << " | Nama: " << nama << " | Merek: " << merek << " | Harga: " << harga << endl;
    }
};

File: PHP/elektronik.php
<?php
class elektronik {
    private $id;
    private $nama;
    private $merek;
    private $harga;
    private $gambar; // ← tambahkan atribut ini

    public function __construct($id, $nama, $merek, $harga, $gambar = '') {
        $this->id = $id;
        $this->nama = $nama;
        $this->merek = $merek;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // --- Getter ---
    public function getId()    { return $this->id; }
    public function getNama()  { return $this->nama; }
    public function getMerek() { return $this->merek; }
    public function getHarga() { return $this->harga; }
    public function getGambar(){ return $this->gambar; }

    // --- Setter ---
    public function setNama($nama)    { $this->nama = $nama; }
    public function setMerek($merek)  { $this->merek = $merek; }
    public function setHarga($harga)  { $this->harga = $harga; }
    public function setGambar($gambar){ $this->gambar = $gambar; }

    public function tampil() {
        echo "<p>ID: {$this->id}, Nama: {$this->nama}, Merek: {$this->merek}, Harga: {$this->harga}</p>";
        if (!empty($this->gambar)) {
            echo "<img src='{$this->gambar}' width='100'>";
        }
    }
}
?>
